oci/layout/oci_transport.go

```diff
@@ -176,16 +179,54 @@ func (ref ociReference) PolicyConfigurationNamespaces() []string {
 // NOTE: If any kind of signature verification should happen, build an UnparsedImage from the value returned by NewImageSource,
 // verify that UnparsedImage, and convert it into a real Image via image.FromUnparsedImage.
 func (ref ociReference) NewImage(ctx *types.SystemContext) (types.Image, error) {
-	src := newImageSource(ref)
+	d, err := ref.getManifestDescriptor()
+	if err != nil {
+		return nil, err
+	}
+	src := newImageSource(ref, d)
```

Comment:
This duplication between `NewImage` and `NewImageSource` rather suggests that `getManifestDescriptor` should be called within `newImageSource` instead of supplied from the outside.

(That still allows `getManifestDescriptor` to be a method on `ociReference` if you want.)